code-challenges.test.js

```diff
@@ -15,47 +19,323 @@
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
+// const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+// const letterA = "a"
+// // Expected output: ["Apple", "Banana", "Orange"]
+// const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+// const letterE = "e"
+// // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+
+
+//Pseudo Code:
+/*
+-Input an array and a single letter(two arguments), the output is an array containing only elements of the argument array that has the particular letter
+-Perform a .filter() to filter out the elements containing the string letter input
+-Use .includes() to check for the string letter in the .filter function.
+-In addition, need to do a red-green-refactor methodolgy, arrayOfWords1 & arrayOfWords2
+*/
+
+//First Step: Perform Red
+
+// describe("filterWords", () => {
+
+//     it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+
+//         expect(filterWords(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+//     })
+// })
+
+// Received an error =>  ReferenceError: filterWords is not defined
+
+
+//Second Step: Make the test pass for arrayOfWords1 & arrayOfWords2
+
+// describe("filterWords", () => {
+
+//     const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+//     const letterA = "a"
+// // Expected output: ["Apple", "Banana", "Orange"]
+//     const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+//     const letterE = "e"
+// // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+//     it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+
+//         expect(filterWords(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"]);
+//         expect(filterWords(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"]);
+//     })
+// })
+
+
+
+// b) Create the function that makes the test pass.
+// -Input an array and a single letter(two arguments), the output is an array containing only elements of the argument array that has the particular letter
+// -Perform a .filter() to filter out the elements containing the string letter input
+// -Use .includes() to check for the string letter in the .filter function.
+
+// const filterWords = (arrayOfWords, letter) => {
+//     newArray = arrayOfWords.filter(value => {
+
+//         return  value.includes(letter) || value.includes(letter.toUpperCase())
+//     })
+//     return newArray
+// }
+
+
+//Third Step: Refactor
+/*
+Reformat by having implicit arrow functions 
+*/
+const filterWords = (arrayOfWords, letter) => arrayOfWords.filter(value => value.includes(letter.toLowerCase()) || value.includes(letter.toUpperCase()))
+
```

Comment:
Nice work. I like the concise approach. I think it is cool to explore the one-liner syntax given by the `=>` but find a balance between code that is concise and code that is readable.